### PR TITLE
Display names of users in options when uploading a spectra

### PR DESCRIPTION
**Pull request by jadalilleboe**
Display a user's full name in the option for adding observers/reducers when uploading a spectra
Screenshot (39)
Screenshot (40)

### CODE REVIEW

**Review by profjsb — APPROVED** [review]

**Comment by profjsb** [comment]
You might check if there's a need to line wrap in the case of very long names/username

File: static/js/components/UploadSpectrum.jsx [hunk 1]
```diff
@@ -313,7 +316,10 @@ const UploadSpectrumForm = ({ route }) => {
           type: "integer",
           anyOf: users?.map((user) => ({
             enum: [user.id],
-            title: user.username,
+            title:
+              user?.first_name && user?.last_name
+                ? `${user.username} (${user.first_name} ${user.last_name})`
+                : `${user.username}`,
```
Comment (stefanv) [inline static/js/components/UploadSpectrum.jsx:322]:
```suggestion
            title: `${user.username} (${first ? first + " " : ""}${last ? last : ""})`,
```

Comment (stefanv) [reply static/js/components/UploadSpectrum.jsx:322]: Same below.

Otherwise, if either first or last name is missing, the real name won't render.

Comment (jadalilleboe) [reply static/js/components/UploadSpectrum.jsx]: fixed!

**Comment by jadalilleboe** [comment]
@profjsb Doesn't seem like there's an easy way to add a line break with rjsf form? looking at [this issue](https://github.com/rjsf-team/react-jsonschema-form/issues/2562)

**Review by stefanv — APPROVED** [review]

**Comment by mcoughlin** [comment]
Given the two approvals, and that react-jsonschema-form does not yet support line breaks, I think we merge this and return to line breaking if we find there are too many long names given trouble here.

**Pull request merged**